### DISCUSSION TITLE
Implement dataset removal method

### DISCUSSION THREAD

**Discussion by justinas-kazanavicius** [langfuse/langfuse#3050] [Ideas]
### Describe the feature or potential improvement

At the moment Python SDK only supports dataset creation, but there are no methods for removing datasets and it has to be done through UI which is tedious when developing dataset creation scripts.

Suggestion: implement `Langfuse.remove_dataset` (or `delete_dataset`) methods.

**Comment by russomario** [comment]
any news on this?

**Comment by musaabhasan** [comment]
A dataset removal method would be useful, but I would separate the API into two explicit operations rather than one ambiguous delete:

1. **Archive / disable dataset**: remove it from normal experiment selection, keep prior run references intact, and preserve auditability.
2. **Hard delete dataset**: remove dataset items and metadata where retention policy allows it, with a clear response showing what was deleted and what was retained.

For evaluation systems, hard deletion can break reproducibility if old experiment runs still point to dataset items. A safer contract could be:

```python
langfuse.datasets.archive(name="rag-regression-v1")
langfuse.datasets.delete(
    name="rag-regression-v1",
    mode="hard",
    require_no_active_runs=True,
    idempotency_key="cleanup-2026-05-08-rag-regression-v1",
)
```

The response should ideally include counts and retained references:

```json
{
  "dataset": "rag-regression-v1",
  "deleted_items": 120,
  "retained_run_references": 4,
  "state": "archived",
  "deletion_receipt_id": "..."
}
```

This matters for governance because datasets often contain prompts, user examples, retrieved context, labels, or evaluator notes. Teams need a programmatic cleanup path, but also an evidence trail showing whether deletion was complete, blocked by experiment lineage, or converted into archival retention.
